addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -197,6 +201,11 @@ function factory(dependencies) {
         isPreparingLoading: attr({
             default: false,
         }),
+        /**
+         * Determines whether `this` should automatically scroll on receiving
+         * a new message.
+         */
+        hasAutoScrollOnMessageReceived: attr(),
```

Comment:
Would clarify that reception of new message is detected by means of component hints.